### DISCUSSION TITLE
New feature request

### DISCUSSION THREAD

**Discussion by valakia** [statisticssweden/PxWeb#292] [Ideas]
Hello! I don't know if this is the correct place to ask about this but let's try. I was wondering about a new feature that could be added to the API.

I have a website that gets its data from the PxWeb API. It shows various statistics in charts. I want to show a chart with the latest employment data and add a second value for comparison that would be from last year. I'm using json-format so I'm given the opportunity to get the data as is or with updates as the new data is released. That would be the desired end result that the chart would update on its own. But when I choose the updating option, the second value from last year gets updated to the latest value. So now both of the values are the same.

So could there be a parameter to be added that could keep the correlation between the two values so that the second value would remain as the previous years value? It should be updated as the newest value updates but would keep the one year period in between. Or even if it would remain as a static non-changing value would be good.

This is a bit tricky to explain but I hope this is understandable. Thanks!

**Comment by likp** [comment]
Unfortunately this is not possible unless you inspect the metadata for the table and rewrite your query. 
We are currently working on a version 2 of the API where we have a more powerful way to express selections. I just want to make sure if we would have your case covered. I guess that you wish for to cases 
1. If I select _January 2022_ I would like to select _January 2021_ and when February values are available the same query would fetch _February 2022_ and _February 2021_.
2. If I select _January 2021_  and _January 2022_ and when February 2022 values comes the same query should return _January 2021_ and _February 2022_.

@valakia Is this correctly understood?

**Comment by valakia** [comment]
Thank you so much for the reply.

Yes the first proposition (1) would work for my case: I choose January 2021 and January 2022 to my chart. And when February 2022 is available, I would like the same query to fetch February 2021 and February 2022.

The second proposition could also be very useful, if it would be available. Actually that would also work with my current case. So either one or both features would be very appreciated. The first proposition would be slightly better for my case if I had to choose just one.

Thanks again!

**Comment by likp** [comment]
Thanks for the feedback. We will probably not do anything to the existing API but try to come up with a solution to make this possible in version 2 of the API.

**Reply by valakia** [reply]
Thank you very much. I'm looking forward for the version 2.
